frontend/src/features/mapSideBar/tabs/propertyDetails/usePropertyDetails.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  ALR_LAYER_URL,
+  ELECTORAL_LAYER_URL,
+  HWY_DISTRICT_LAYER_URL,
+  INDIAN_RESERVES_LAYER_URL,
+  IUserLayerQuery,
+  MOTI_REGION_LAYER_URL,
+  useLayerQuery,
+} from 'components/maps/leaflet/LayerPopup';
+import { useProperties } from 'hooks';
+import useIsMounted from 'hooks/useIsMounted';
+import { LatLngLiteral } from 'leaflet';
+import { useEffect, useMemo, useState } from 'react';
+import { pidFormatter } from 'utils';
+
+import { IPropertyDetailsForm, toFormValues } from './PropertyDetailsTabView.helpers';
+
+export function usePropertyDetails(pid?: string) {
+  const { getPropertyWithPid } = useProperties();
+  const isMounted = useIsMounted();
+  const [propertyDetails, setPropertyDetails] = useState<IPropertyDetailsForm | undefined>(
+    undefined,
+  );
+
+  const lat = propertyDetails?.latitude;
+  const lng = propertyDetails?.longitude;
+  const location = useMemo(
+    () => (lat === undefined || lng === undefined ? undefined : { lat, lng }),
+    [lat, lng],
+  );
+
+  const motiRegionService = useLayerQuery(MOTI_REGION_LAYER_URL);
+  const highwaysDistrictService = useLayerQuery(HWY_DISTRICT_LAYER_URL);
+  const electoralService = useLayerQuery(ELECTORAL_LAYER_URL);
+  const alrService = useLayerQuery(ALR_LAYER_URL);
+  const firstNationsService = useLayerQuery(INDIAN_RESERVES_LAYER_URL);
+
+  useEffect(() => {
+    async function fn() {
+      if (!pid) {
+        return;
+      }
+      const propInfo = await getPropertyWithPid(pid);
+      if (isMounted() && propInfo.pid === pidFormatter(pid)) {
+        setPropertyDetails(toFormValues(propInfo));
+      }
+    }
+
+    fn();
+  }, [getPropertyWithPid, isMounted, pid]);
+
+  useEffect(() => {
+    async function fn() {
+      if (location === undefined) {
+        return;
+      }
+      // query BC Geographic Warehouse layers
+      const motiRegion = await findByLocation(motiRegionService, location, 'GEOMETRY');
```

Comment:
are there any requirements on error handling for these?

Reply:
the underlying custom axios that queries map layers handles errors. The story does not call for specific error action (that I can see)